var/spack/repos/builtin/packages/openpmd-api/package.py

```diff
@@ -24,27 +24,29 @@ class OpenpmdApi(CMakePackage):
             description='Enable ADIOS1 support')
     variant('adios2', default=False,
             description='Enable ADIOS2 support')
-    # variant('json', default=True,
-    #         description='Enable JSON support')
-    variant('python', default=True,
+    variant('json', default=True,
+            description='Enable JSON support')
+    variant('python', default=False,
             description='Enable Python bindings')
 
-    depends_on('cmake@3.10.0:', type='build')
+    depends_on('cmake@3.11.0:', type='build')
     depends_on('mpark-variant@1.3.0:')
     depends_on('catch@2.3.0: ~single_header', type='test')
     depends_on('mpi@2.3:', when='+mpi')  # might become MPI 3.0+
     depends_on('hdf5@1.8.13:', when='+hdf5')
     depends_on('hdf5@1.8.13: ~mpi', when='~mpi +hdf5')
     depends_on('hdf5@1.8.13: +mpi', when='+mpi +hdf5')
-    depends_on('adios@1.10.0:', when='+adios1')
-    depends_on('adios@1.10.0: ~mpi', when='~mpi +adios1')
-    depends_on('adios@1.10.0: +mpi', when='+mpi +adios1')
-    depends_on('adios2@2.1.0:', when='+adios2')
-    depends_on('adios2@2.1.0: ~mpi', when='~mpi +adios2')
-    depends_on('adios2@2.1.0: +mpi', when='+mpi +adios2')
+    depends_on('adios@1.13.1:', when='+adios1')
+    depends_on('adios@1.13.1: ~mpi', when='~mpi +adios1')
+    depends_on('adios@1.13.1: +mpi', when='+mpi +adios1')
+    depends_on('adios2@2.3.0:', when='+adios2')
+    depends_on('adios2@2.3.0: ~mpi', when='~mpi +adios2')
+    depends_on('adios2@2.3.0: +mpi', when='+mpi +adios2')
+    depends_on('nlohmann-json@3.4.0:', when='+json')
     # ideally we want 2.3.0+ for full C++11 CT function signature support
     depends_on('py-pybind11@2.2.3:', when='+python')
     depends_on('py-numpy@1.15.1:', when='+python', type=['test', 'run'])
+    depends_on('python@3.5:', when='+python')
```

Comment:
`type=('build', 'run')`? Same for the previous two dependencies?

Reply:
good points!
- pybind11 should be build only (header-only)
- numpy's C API is re-implemented in pybind11 and should stay test/run for the actual python module
- python should be build, test, run for C API and runtime